crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -819,6 +842,19 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
             ));
         }
 
+        for ((update, (_, table_uuid)), response) in updates
```

Comment:
I think we should push the whole block up to right after C::commit_table_transaction.
There is no need to load storage secret if the contract check fails.

Reply:
done